WebSocket/src/wsQueryInterface.ts

```diff
@@ -0,0 +1,174 @@
+        this._wsQueryClient = new TDWebSocketClient(this._url);
+    }
+
+    connect(database?: string): Promise<WSConnResponse> {
+
+        let _db = this._url.pathname.split('/')[3];
+
+        if (database) {
+            _db = database;
+        }
+        this._reqIDIncrement()
+        let connMsg = {
+            action: 'conn',
+            args: {
+                req_id: this._req_id,
+                user: this._url.username,
+                password: this._url.password,
+                db: _db,
+            }
+        }
+        return new Promise((resolve, reject) => {
+            this._wsQueryClient.Ready()
+                .then((ws: TDWebSocketClient) => {
+                    return ws.sendMsg(JSON.stringify(connMsg))
+                })
+                .then((e: any) => {
+                    if (e.code == 0) {
+                        resolve(e);
+                    } else {
+                        reject(new WebSocketQueryError(`${e.message}, code ${e.code}`))
+                    }
+                })
+        })
+    }
+
+    // need to construct Response.
+    query(sql: string): Promise<WSQueryResponse> {
+        this._reqIDIncrement()
+        // construct msg
+        let queryMsg = {
+            action: 'query',
+            args: {
+                req_id: this._req_id,
+                sql: sql
+            },
+        }
+        return new Promise((resolve, reject) => {
+            this._wsQueryClient.sendMsg(JSON.stringify(queryMsg))
+                .then((e: any) => {
+                    if (e.code == 0) {
+                        resolve(new WSQueryResponse(e))
+                    } else {
+                        reject(new WebSocketInterfaceError(`${e.message},code ${e.code}`))
+                    }
+                })
+        })
+    }
+
+    getState() {
+        return this._wsQueryClient.readyState()
+    }
+
+    fetch(res: WSQueryResponse): Promise<WSFetchResponse> {
+        this._reqIDIncrement()
+        let fetchMsg = {
+            action: 'fetch',
+            args: {
+                req_id: this._req_id,
+                id: res.id
+            }
+        }
+        return new Promise((resolve, reject) => {
+            this._wsQueryClient.sendMsg(JSON.stringify(fetchMsg)).then((e: any) => {
+                if (e.code == 0) {
+                    resolve(new WSFetchResponse(e))
+                } else {
+                    reject(new WebSocketInterfaceError(`${e.message},code ${e.code}`))
+                }
+            }).catch(e => {
+                reject(e)
+            });
+        })
+    }
+
+    fetchBlock(fetchResponse: WSFetchResponse, taosResult: TaosResult): Promise<TaosResult> {
+        this._reqIDIncrement()
+        let fetchBlockMsg = {
+            action: 'fetch_block',
+            args: {
+                'req_id': this._req_id,
+                'id': fetchResponse.id,
+            }
+        }
+        return new Promise((resolve, reject) => {
```

Comment:
There're two kind of responses for fetchBlock:

1. binary data
2. json with error message like other actions.

Reply:
fix in next pr